src/app/tabs/runTab/settings.js

```diff
@@ -212,6 +212,7 @@ class SettingsUI {
     this.event.trigger('clearInstance', [])
     this.updateNetwork()
     this.updatePlusButton()
+    this.blockchain.event.trigger('newExecutionContext', [])
```

Comment:
`this.blockchain` should already trigger the event `contextChanged`, you don't need this one.

Reply:
Yes, `this.blockchain.event.on('contextChanged', () => {` should have been a better option for listening to an already existing event, but the EventManager which `this.blockchain` uses from remix-lib does not support the `.on('existingEvent')`.

Reply:
It only has support for `register`, `unregister` and `trigger` event.

Reply:
that's because of some inconsistency in the 2 event libs we use.....
could you use `register` ? `this.blockchain.event.register('contextChanged' , () => {})`

Reply:
Will this not override the already existing callback for `'contextChanged'` event? I'll give it a try, i hope it does not break anything.

Reply:
is this working  `this.blockchain.executionContext.event.register('contextChanged'`  ?
that's not ideal, but at least we don't add another event

Reply:
Using `this.blockchain.event.register('contextChanged')` or `this.blockchain.executionContext.event.register('contextChanged')` fires the event, but the callback is not called because somehow `this.networkcallid > callid` in **blockchain.js**
```javascript
  updateNetwork (cb) {
    this.networkcallid++
    ((callid) => {
      this.executionContext.detectNetwork((err, { id, name } = {}) => {
        if (this.networkcallid > callid) return //<--- This line
        this.networkcallid++
        if (err) {
          return cb(err)
        }
        cb(null, {id, name})
      })
    })(this.networkcallid)
  }
```